lib/pegboard/container-runner/src/main.rs

```diff
@@ -0,0 +1,348 @@
+use std::{
+	fs,
+	io::{BufRead, BufReader},
+	path::Path,
+	process::{Command, Stdio},
+	sync::mpsc,
+	thread,
+	time::{Duration, SystemTime, UNIX_EPOCH},
+};
+
+use anyhow::*;
+use signal_hook::{consts::signal::SIGTERM, iterator::Signals};
+use utils::{var, Stakeholder};
+
+pub mod log_shipper;
+pub mod throttle;
+pub mod utils;
+
+/// Maximum length of a single log line
+const MAX_LINE_BYTES: usize = 1024;
+
+/// Maximum number of bytes to buffer before dropping logs
+const MAX_BUFFER_BYTES: usize = 1024 * 1024;
+
+/// Maximum number of lines to print to stdout for debugging. This helps
+/// identify the reasons for program crashes from the container's output.
+const MAX_PREVIEW_LINES: usize = 128;
+
+fn main() -> anyhow::Result<()> {
```

Comment:
we should have a main_inner that catches errors and writes it to a file in order to catch container-runner errors